### PR TITLE
feat: move existing BOM generation to `make-bom` subcommand

### PR DESCRIPTION
**Pull request by madpah**
Implementation of #527.

### CODE REVIEW

File: cyclonedx_py/command/make_bom.py [hunk 1]
```diff
@@ -0,0 +1,256 @@
...
+        try:
+            parser = self._get_input_parser()
+        except CycloneDxCmdNoInputFileSupplied as error:
+            print(f'ERROR: {str(error)}', file=sys.stderr)
+            exit(1)
+        except CycloneDxCmdException as error:
+            print(f'ERROR: {str(error)}', file=sys.stderr)
+            exit(1)
+
+        if parser and parser.has_warnings():
```
[the remark this answers is not in the document]
Comment (madpah) [reply cyclonedx_py/command/make_bom.py]: Resolved

File: cyclonedx_py/command/make_bom.py [hunk 2]
```diff
@@ -0,0 +1,256 @@
...
+                  '!!                                                      !!',
+                  *('!! -> {} !!'.format(warning.get_item().ljust(49)) for warning in parser.get_warnings()),
+                  '!!                                                      !!',
+                  '!! The above will NOT be included in the generated      !!',
+                  '!! CycloneDX as version is a mandatory field.           !!',
+                  '!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!!',
+                  '',
+                  sep='\n', file=sys.stderr)
+
+        bom = Bom.from_parser(parser=parser)
```
[the remark this answers is not in the document]
Comment (madpah) [reply cyclonedx_py/command/make_bom.py]: Resolved.

File: cyclonedx_py/command/make_bom.py [hunk 3]
```diff
@@ -0,0 +1,265 @@
...
+                debug_message=lambda m, *a, **k: self._debug_message(f'CondaListExplicitParser {m}', *a, **k)
+            )
+        elif self.arguments.input_from_conda_json:
+            return CondaListJsonParser(
+                conda_data=input_data,
+                use_purl_bom_ref=self.arguments.use_purl_bom_ref,
+                debug_message=lambda m, *a, **k: self._debug_message(f'CondaListJsonParser {m}', *a, **k)
+            )
+        elif self.arguments.input_from_pip:
+            return PipEnvParser(
```
[the remark this answers is not in the document]
Comment (madpah) [reply cyclonedx_py/command/make_bom.py:188]: Out of date and Lift not updating??

@sonatype-lift ignore

File: cyclonedx_py/command/make_bom.py [hunk 4]
```diff
@@ -0,0 +1,265 @@
...
+                pipenv_contents=input_data,
+                use_purl_bom_ref=self.arguments.use_purl_bom_ref,
+                debug_message=lambda m, *a, **k: self._debug_message(f'PipEnvParser {m}', *a, **k)
+            )
+        elif self.arguments.input_from_poetry:
+            return PoetryParser(
```
[the remark this answers is not in the document]
Comment (madpah) [reply cyclonedx_py/command/make_bom.py:194]: Out of date and Lift not updating??

@sonatype-lift ignore

**Comment by jkowalleck** [comment]
ready for review? will have a look tomorrow.

**Comment by jkowalleck** [comment]
outdated.

**Pull request closed without merging**